File: practice_4-25/Prac.java
import java.util.ArrayList;

public class Prac {
    public static void main(String[] args) {
        ArrayList<String> strArr = new ArrayList<String>();
        strArr.add("a");
        strArr.add("b");
        strArr.add("c");
        strArr.add("a");
        strArr.add("b");
        strArr.add("b");
        strArr.add("a");
        System.out.println(computeArray(strArr));
    }

    public static int computeArray(ArrayList<String> strArr) {
        ArrayList<String> newArr = new ArrayList<String>();
        int matches = 0;
        // for (String s : strArr) {
        // if (newArr.contains(s)) {
        // newArr.remove(s);
        // matches++;
        // } else {
        // newArr.add(s);
        // }
        // }
        int i = 0;
        while (i < strArr.size()) {
            if (newArr.contains(strArr.get(i))) {
                newArr.remove(strArr.get(i));
                matches++;
            } else {
                newArr.add(strArr.get(i));
            }
            i++;
        }
        return matches;
    }
}

File: practice_4-25/prac.py
def compute_list(str_list):
    new_list = []
    matches = 0
    # for s in str_list:
    #     if s in new_list:
    #         new_list.remove(s)
    #         matches += 1
    #     else:
    #         new_list.append(s)
    for s in str_list:
        new_list.append(s)
        if len(set(new_list)) != len(new_list):
            matches += 1
    # i = 0
    # while i < len(str_list):
    #     if str_list[i] in new_list:
    #         new_list.remove(str_list[i])
    #         matches += 1
    #     else:
    #         new_list.append(str_list[i])
    print(str(matches))


def main():
    str_list = ["a", "b", "c", "b", "b", "a", "a"]
    compute_list(str_list)


if __name__ == "__main__":
    main()
